### DISCUSSION TITLE
Code to input simulation into MoBi

### DISCUSSION THREAD

**Discussion by rachudso** [Open-Systems-Pharmacology/Forum#1445] [Questions & Problems]
Hello, 

Does anyone know of a R code available to input a simulation into MoBi by specifying each building block? For example, I have simulations I need to run for 70+ different administrations (these are included in the events building block). I currently have all these event building blocks saved as a pkml file. It would be very convenient to have an R code that specifies what event pkml file to use and input each simulation into MoBi (instead of having to add each of these simulations by hand in MoBi). Is this possible?

For reference, I did download the ospsuite-R package. However, it seems this is for running a simulation in R, not inputting building block files into MoBi to run the simulation in MoBi.

**Comment by AndreDlm** [comment]
Hi @rachudso,

I don't have a R code at hand that I could share, but this shouldn't be too difficult, depending of course on the administration types you have. What I would propose conceptually is:
* Set up a simulation with a placeholder for your administration(s), e.g. _i_ doses of zero drug amount at _j_ times
* Export the simulation to pkml and load it in R
* Create an Excel list defining for each administration protocol, the number (_i_) and times (_j_) of the doses as well as the drug amount of each dose
* In R, loop over the Excel list and with each iteration, access the Administration Protocol parameters (https://www.open-systems-pharmacology.org/OSPSuite-R/articles/load-get.html) and set the parameter values according to the data in your Excel table (https://www.open-systems-pharmacology.org/OSPSuite-R/articles/set-values.html)

Alternatively, and more efficiently than the loop, you could use the _SimulationBatch_ functions described [here](https://www.open-systems-pharmacology.org/OSPSuite-R/articles/efficient-calculations.html#optimized-loading-for-running-simulations-with-predefined-set-of-variable-parameters)

I hope this helps,
André
